File: SPRING_BOOT/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/EmergencyBedService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.dto.request.EmergencyBedRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.response.EmergencyBedResponse;

import java.util.List;
import java.util.Map;

public interface EmergencyBedService {

    EmergencyBedResponse create(EmergencyBedRequest request);

    EmergencyBedResponse getById(Long id);

    List<EmergencyBedResponse> getAll();

    EmergencyBedResponse updateStatus(Long id, String status);

    EmergencyBedResponse assignBed(Long bedId, Long emergencyPatientId);

    EmergencyBedResponse releaseBed(Long bedId);

    List<EmergencyBedResponse> getByStatus(String status);

    List<EmergencyBedResponse> getByWardName(String wardName);

    Long getAvailableCount();

    Map<String, Object> getWardSummary();
}

File: SPRING_BOOT_Hospital_Management/Hospital_Management_system_angular/src/app/services/emergency/emergency-bed.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { EmergencyBed } from '../../models/emergency';

@Injectable({ providedIn: 'root' })
export class EmergencyBedService {
  private api = environment.apiUrl + 'emergency/beds';

  constructor(private http: HttpClient) {}

  create(bed: EmergencyBed): Observable<EmergencyBed> {
    return this.http.post<EmergencyBed>(this.api, bed);
  }

  getById(id: number): Observable<EmergencyBed> {
    return this.http.get<EmergencyBed>(`${this.api}/${id}`);
  }

  getAll(): Observable<EmergencyBed[]> {
    return this.http.get<EmergencyBed[]>(this.api);
  }

  updateStatus(id: number, status: string): Observable<EmergencyBed> {
    return this.http.put<EmergencyBed>(`${this.api}/${id}/status`, { status });
  }

  assignBed(bedId: number, emergencyPatientId: number): Observable<EmergencyBed> {
    return this.http.put<EmergencyBed>(`${this.api}/${bedId}/assign`, { emergencyPatientId });
  }

  releaseBed(id: number): Observable<EmergencyBed> {
    return this.http.put<EmergencyBed>(`${this.api}/${id}/release`, {});
  }

  getByStatus(status: string): Observable<EmergencyBed[]> {
    return this.http.get<EmergencyBed[]>(`${this.api}/status/${status}`);
  }

  getByWardName(wardName: string): Observable<EmergencyBed[]> {
    return this.http.get<EmergencyBed[]>(`${this.api}/ward/${wardName}`);
  }

  getAvailableCount(): Observable<number> {
    return this.http.get<number>(`${this.api}/available-count`);
  }

  getWardSummary(): Observable<any> {
    return this.http.get<any>(`${this.api}/ward-summary`);
  }
}
